chainerx_cc/chainerx/routines/connection.cc

```diff
@@ -393,6 +394,28 @@ Array Linear(const Array& x, const Array& w, const absl::optional<Array>& b, uin
     return out_matrix.Reshape(out_shape);
 }
 
+Array EmbedId(const Array& x, const Array& w, absl::optional<int64_t> ignore_label) {
+    if (ignore_label.has_value() && (*ignore_label < 0 || *ignore_label >= w.shape()[0])) {
+        throw IndexError{"ignore_label should be greater than equal to 0 and less than" + w.shape()[0]};
+    }
+    if (ignore_label.has_value()) {
+        Array ignore_label_array = Full(x.shape(), *ignore_label, x.dtype(), x.device());
+        Array mask = Where(x == ignore_label_array, -1, x);
```

Comment:
Why cannot this fill value be 0 (like `F.embed_id`)?

Reply:
When `x` has an index 0, then it will end up zeroing that as well when it should actually be the actual vector in 0th row of `W`.

Reply:
Ah, OK.
I think the `mask` only has to have `bool` dtype.
I.e. `Array mask = x == ignore_label_array;` and `Array out = Where(mask_, 0, Take(w, Where(_mask, 0, x), 0));`.